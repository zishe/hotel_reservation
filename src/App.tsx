import React, { Component } from "react";
import { AppNavigationContainer } from "./routes/AppStackNavigator";
import { ApolloProvider } from "react-apollo";
import { Root } from "native-base";
import { apolloClient } from "./graphql/client/apolloClient";

export default class App extends Component {
  render() {
    return (
      <ApolloProvider client={apolloClient}>
        <Root>
          <AppNavigationContainer />
        </Root>
      </ApolloProvider>
    );
  }
}
