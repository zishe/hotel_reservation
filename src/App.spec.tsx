import "react-native";
import React from "react";
import App from "./App";
import renderer from "react-test-renderer";

jest.useFakeTimers();
describe("App", () => {
  it("renders correctly", () => {
    const app = renderer.create(<App />).toJSON();
    expect(app).toMatchSnapshot();
  });
});
