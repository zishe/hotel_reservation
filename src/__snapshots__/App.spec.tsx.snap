// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={false}
    onGestureHandlerEvent={[Function]}
    onGestureHandlerStateChange={[Function]}
    onLayout={[Function]}
    style={
      Object {
        "flex": 1,
        "overflow": "hidden",
        "zIndex": 0,
      }
    }
  >
    <View
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <View
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <View
          collapsable={false}
          pointerEvents="auto"
          removeClippedSubviews={false}
          style={
            Array [
              Object {
                "flex": 1,
                "overflow": "hidden",
              },
              Array [
                Object {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                },
                Object {
                  "opacity": 1,
                },
              ],
            ]
          }
        >
          <View
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "backgroundColor": "#fff",
                  "flex": 1,
                  "height": 1334,
                }
              }
            >
              <RCTScrollView
                automaticallyAdjustContentInsets={false}
                contentContainerStyle={
                  Array [
                    Object {
                      "padding": undefined,
                    },
                    Object {
                      "alignItems": "stretch",
                      "backgroundColor": "#F5FCFF",
                      "flex": 1,
                      "justifyContent": "flex-start",
                    },
                  ]
                }
                contentInset={
                  Object {
                    "bottom": 0,
                  }
                }
                enableAutomaticScroll={true}
                enableOnAndroid={false}
                enableResetScrollToCoords={true}
                extraHeight={75}
                extraScrollHeight={0}
                getScrollResponder={[Function]}
                handleOnScroll={[Function]}
                keyboardDismissMode="interactive"
                keyboardOpeningTime={250}
                keyboardShouldPersistTaps="handled"
                keyboardSpace={0}
                onScroll={[Function]}
                resetKeyboardSpace={[Function]}
                resetScrollToCoords={
                  Object {
                    "x": 0,
                    "y": 0,
                  }
                }
                scrollEventThrottle={1}
                scrollForExtraHeightOnAndroid={[Function]}
                scrollIntoView={[Function]}
                scrollToEnd={[Function]}
                scrollToFocusedInput={[Function]}
                scrollToPosition={[Function]}
                showsVerticalScrollIndicator={true}
                style={
                  Object {
                    "backgroundColor": "transparent",
                    "flex": 1,
                  }
                }
                update={[Function]}
                viewIsInsideTabBar={false}
              >
                <View>
                  <View
                    onLayout={[Function]}
                  >
                    <View
                      androidStatusBarColor="lightgray"
                      iosBarStyle="dark-content"
                      style={
                        Array [
                          Object {
                            "backgroundColor": "#F8F8F8",
                            "borderBottomColor": "#a7a6ab",
                            "borderBottomWidth": 0.5,
                            "elevation": 3,
                            "flexDirection": "row",
                            "height": 64,
                            "justifyContent": "center",
                            "left": 0,
                            "paddingLeft": 6,
                            "paddingRight": 10,
                            "paddingTop": 18,
                            "right": 0,
                            "shadowColor": undefined,
                            "shadowOffset": undefined,
                            "shadowOpacity": undefined,
                            "shadowRadius": undefined,
                            "top": 0,
                          },
                          Object {
                            "backgroundColor": "#104c97",
                            "elevation": 4,
                            "shadowColor": "black",
                            "shadowOpacity": 1,
                          },
                        ]
                      }
                    >
                      <View
                        style={
                          Array [
                            Object {
                              "alignItems": "flex-start",
                              "alignSelf": "center",
                              "flex": 1,
                            },
                            Object {
                              "flex": 2,
                            },
                          ]
                        }
                      >
                        <Text
                          allowFontScaling={false}
                          onPress={[Function]}
                          style={
                            Array [
                              Object {
                                "color": undefined,
                                "fontSize": 12,
                              },
                              Array [
                                Object {
                                  "color": "#000",
                                  "fontSize": 30,
                                },
                                Object {
                                  "color": "#000",
                                  "fontSize": 30,
                                },
                                Object {
                                  "color": "#fff",
                                  "padding": 5,
                                },
                              ],
                              Object {
                                "fontFamily": "Ionicons",
                                "fontStyle": "normal",
                                "fontWeight": "normal",
                              },
                              Object {},
                            ]
                          }
                          testID="menu"
                        >
                          
                        </Text>
                      </View>
                      <View
                        style={
                          Array [
                            Object {
                              "alignItems": "center",
                              "alignSelf": "center",
                              "flex": 1,
                            },
                            Object {
                              "alignItems": "center",
                              "flex": 4,
                            },
                          ]
                        }
                      >
                        <Text
                          numberOfLines={1}
                          style={
                            Array [
                              Object {
                                "color": "#000",
                                "fontFamily": "System",
                                "fontSize": 17,
                                "fontWeight": "700",
                                "marginLeft": undefined,
                                "paddingLeft": 4,
                                "paddingTop": 1,
                                "textAlign": "center",
                              },
                              Object {
                                "color": "#fff",
                              },
                            ]
                          }
                        >
                          Hotel Reservations
                        </Text>
                      </View>
                      <View
                        style={
                          Array [
                            Object {
                              "alignItems": "flex-end",
                              "alignSelf": "center",
                              "flex": 1,
                              "flexDirection": "row",
                              "justifyContent": "flex-end",
                            },
                            Object {
                              "flex": 2,
                            },
                          ]
                        }
                      >
                        <Text
                          allowFontScaling={false}
                          style={
                            Array [
                              Object {
                                "color": undefined,
                                "fontSize": 12,
                              },
                              Array [
                                Object {
                                  "color": "#000",
                                  "fontSize": 30,
                                },
                                Object {
                                  "color": "#000",
                                  "fontSize": 30,
                                },
                                Object {
                                  "color": "#fff",
                                  "padding": 5,
                                },
                              ],
                              Object {
                                "fontFamily": "Ionicons",
                                "fontStyle": "normal",
                                "fontWeight": "normal",
                              },
                              Object {},
                            ]
                          }
                          testID="contact"
                        >
                          
                        </Text>
                      </View>
                    </View>
                  </View>
                  <View
                    style={
                      Object {
                        "alignItems": "stretch",
                        "backgroundColor": "#F5FCFF",
                        "flex": 8,
                        "justifyContent": "flex-start",
                      }
                    }
                  >
                    <ActivityIndicator
                      animating={true}
                      color="#00ff00"
                      hidesWhenStopped={true}
                      size="large"
                      testID="activityIndicator"
                    />
                  </View>
                </View>
              </RCTScrollView>
            </View>
          </View>
        </View>
      </View>
      <View
        collapsable={false}
        onGestureHandlerEvent={[Function]}
        onGestureHandlerStateChange={[Function]}
        pointerEvents="none"
        style={
          Object {
            "backgroundColor": "black",
            "bottom": 0,
            "left": 0,
            "opacity": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "zIndex": 1000,
          }
        }
      />
    </View>
    <View
      accessibilityViewIsModal={false}
      pointerEvents="box-none"
      style={
        Object {
          "bottom": 0,
          "flexDirection": "row",
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "transform": Array [
            Object {
              "translateX": -320,
            },
          ],
          "zIndex": 1001,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "flex": 0,
            },
            Object {
              "backgroundColor": "white",
              "width": 320,
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "flex": 1,
              },
              undefined,
            ]
          }
        >
          <RCTScrollView
            alwaysBounceVertical={false}
          >
            <View>
              <View
                onLayout={[Function]}
                pointerEvents="box-none"
                style={
                  Object {
                    "paddingBottom": 0,
                    "paddingLeft": 0,
                    "paddingRight": 0,
                    "paddingTop": 20,
                  }
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "paddingVertical": 4,
                      },
                      undefined,
                    ]
                  }
                >
                  <View
                    accessibilityLabel="Reservatinos"
                    accessible={true}
                    isTVSelectable={true}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Object {
                        "opacity": 1,
                      }
                    }
                  >
                    <View
                      onLayout={[Function]}
                      pointerEvents="box-none"
                      style={
                        Object {
                          "backgroundColor": "rgba(0, 0, 0, .04)",
                          "paddingBottom": 0,
                          "paddingLeft": 0,
                          "paddingRight": 0,
                          "paddingTop": 0,
                        }
                      }
                    >
                      <View
                        style={
                          Array [
                            Object {
                              "alignItems": "center",
                              "flexDirection": "row",
                            },
                            undefined,
                          ]
                        }
                      >
                        <View
                          style={
                            Array [
                              Object {
                                "alignItems": "center",
                                "marginHorizontal": 16,
                                "width": 24,
                              },
                              null,
                              undefined,
                            ]
                          }
                        >
                          <Text
                            allowFontScaling={false}
                            style={
                              Array [
                                Object {
                                  "color": undefined,
                                  "fontSize": 12,
                                },
                                Array [
                                  Object {
                                    "color": "#000",
                                    "fontSize": 30,
                                  },
                                  Object {
                                    "color": "#000",
                                    "fontSize": 30,
                                  },
                                ],
                                Object {
                                  "fontFamily": "Ionicons",
                                  "fontStyle": "normal",
                                  "fontWeight": "normal",
                                },
                                Object {},
                              ]
                            }
                          >
                            
                          </Text>
                        </View>
                        <Text
                          style={
                            Array [
                              Object {
                                "fontWeight": "bold",
                                "margin": 16,
                              },
                              Object {
                                "color": "#2196f3",
                              },
                              undefined,
                              undefined,
                            ]
                          }
                        >
                          Reservatinos
                        </Text>
                      </View>
                    </View>
                  </View>
                  <View
                    accessible={true}
                    isTVSelectable={true}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Object {
                        "opacity": 1,
                      }
                    }
                  >
                    <View
                      onLayout={[Function]}
                      pointerEvents="box-none"
                      style={
                        Object {
                          "backgroundColor": "transparent",
                          "paddingBottom": 0,
                          "paddingLeft": 0,
                          "paddingRight": 0,
                          "paddingTop": 0,
                        }
                      }
                    >
                      <View
                        style={
                          Array [
                            Object {
                              "alignItems": "center",
                              "flexDirection": "row",
                            },
                            undefined,
                          ]
                        }
                      />
                    </View>
                  </View>
                  <View
                    accessibilityLabel="Make New Reservation"
                    accessible={true}
                    isTVSelectable={true}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Object {
                        "opacity": 1,
                      }
                    }
                  >
                    <View
                      onLayout={[Function]}
                      pointerEvents="box-none"
                      style={
                        Object {
                          "backgroundColor": "transparent",
                          "paddingBottom": 0,
                          "paddingLeft": 0,
                          "paddingRight": 0,
                          "paddingTop": 0,
                        }
                      }
                    >
                      <View
                        style={
                          Array [
                            Object {
                              "alignItems": "center",
                              "flexDirection": "row",
                            },
                            undefined,
                          ]
                        }
                      >
                        <View
                          style={
                            Array [
                              Object {
                                "alignItems": "center",
                                "marginHorizontal": 16,
                                "width": 24,
                              },
                              Object {
                                "opacity": 0.62,
                              },
                              undefined,
                            ]
                          }
                        >
                          <Text
                            allowFontScaling={false}
                            style={
                              Array [
                                Object {
                                  "color": undefined,
                                  "fontSize": 12,
                                },
                                Array [
                                  Object {
                                    "color": "#000",
                                    "fontSize": 30,
                                  },
                                  Object {
                                    "color": "#000",
                                    "fontSize": 30,
                                  },
                                ],
                                Object {
                                  "fontFamily": "Ionicons",
                                  "fontStyle": "normal",
                                  "fontWeight": "normal",
                                },
                                Object {},
                              ]
                            }
                          >
                            
                          </Text>
                        </View>
                        <Text
                          style={
                            Array [
                              Object {
                                "fontWeight": "bold",
                                "margin": 16,
                              },
                              Object {
                                "color": "rgba(0, 0, 0, .87)",
                              },
                              undefined,
                              undefined,
                            ]
                          }
                        >
                          Make New Reservation
                        </Text>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </RCTScrollView>
        </View>
      </View>
    </View>
  </View>
  <Modal
    animationType="fade"
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    transparent={true}
    visible={false}
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "rgba(0,0,0,0.4)",
          "flex": 1,
          "justifyContent": "flex-end",
          "opacity": 1,
        }
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#fff",
            "elevation": 4,
            "height": 0,
            "maxHeight": 667,
            "minHeight": 56,
            "opacity": 1,
            "padding": 15,
          }
        }
      >
        <RCTScrollView
          data={Array []}
          disableVirtualization={false}
          getItem={[Function]}
          getItemCount={[Function]}
          horizontal={false}
          initialNumToRender={10}
          keyExtractor={[Function]}
          maxToRenderPerBatch={10}
          numColumns={1}
          onContentSizeChange={[Function]}
          onEndReachedThreshold={2}
          onLayout={[Function]}
          onMomentumScrollEnd={[Function]}
          onScroll={[Function]}
          onScrollBeginDrag={[Function]}
          onScrollEndDrag={[Function]}
          renderItem={[Function]}
          scrollEventThrottle={50}
          stickyHeaderIndices={Array []}
          style={
            Object {
              "marginHorizontal": -15,
              "marginTop": 0,
            }
          }
          updateCellsBatchingPeriod={50}
          viewabilityConfigCallbackPairs={Array []}
          windowSize={21}
        >
          <View />
        </RCTScrollView>
      </View>
    </View>
  </Modal>
</View>
`;
